signimage/signimage.go

```diff
@@ -0,0 +1,357 @@
+		panic(0)
+	}
+}
+
+func addToTarball(tw *tar.Writer, filename string, header *tar.Header) error {
+	finfo, err := os.Stat(filename)
+	if err != nil {
+		return fmt.Errorf("failed to stat %s: %w", filename, err)
+	}
+
+	hdr := &tar.Header{
+		Name:     header.Name,
+		Mode:     header.Mode,
+		Typeflag: header.Typeflag,
+		Size:     finfo.Size(),
+	}
+
+	if err := tw.WriteHeader(hdr); err != nil {
+		return fmt.Errorf("failed to write tar header: %w", err)
+	}
+
+	f, err := os.Open(filename)
+	if err != nil {
+		return err
+	}
+	if _, err := io.Copy(tw, f); err != nil {
+		return fmt.Errorf("failed to read file into the tar: %w", err)
+	}
+	f.Close()
+
+	return nil
+
+}
+
+func getAuthFromFile(configfile string, repo string) (authn.Authenticator, error) {
+
+	if configfile == "" {
+		fmt.Printf("\n== no pull secret defined, default to Anonymous\n")
+		return authn.Anonymous, nil
+	}
+
+	f, err := os.Open(configfile)
+	if err != nil {
+		return nil, err
+	}
+	defer f.Close()
+	cf, err := config.LoadFromReader(f)
+	if err != nil {
+		return nil, err
+	}
+
+	var cfg dockertypes.AuthConfig
+
+	cfg, err = cf.GetAuthConfig(repo)
+	if err != nil {
+		return nil, err
+	}
+
+	return authn.FromConfig(authn.AuthConfig{
+		Username:      cfg.Username,
+		Password:      cfg.Password,
+		Auth:          cfg.Auth,
+		IdentityToken: cfg.IdentityToken,
+		RegistryToken: cfg.RegistryToken,
+	}), nil
+
+}
+
+func main() {
+	// get the env vars we are using for setup, or set some sensible defaults
+
+	var unsignedimagename string
+	var signedimagename string
+	var fileslist string
+	var privkeyfile string
+	var pubkeyfile string
+	var pullsecret string
+	var pushsecret string
+	var nopush bool
+	flag.StringVar(&unsignedimagename, "unsignedimage", "", "name of the image to sign")
+	flag.StringVar(&signedimagename, "signedimage", "", "name of the signed image to produce")
+	flag.StringVar(&fileslist, "filestosign", "", "colon seperated list of kmods to sign")
+	flag.StringVar(&privkeyfile, "key", "", "path to file containing private key for signing")
+	flag.StringVar(&pubkeyfile, "cert", "", "path to file containing public key for signing")
+	flag.StringVar(&pullsecret, "pullsecret", "", "path to file containing credentials for pulling images")
+	flag.StringVar(&pullsecret, "pushsecret", "", "path to file containing credentials for pushing images")
+	flag.BoolVar(&nopush, "no-push", false, "do not push the resulting image")
+
+	flag.Parse()
+
+	checkarg(&unsignedimagename, "UNSIGNEDIMAGE", "")
+	checkarg(&signedimagename, "SIGNEDIMAGE", unsignedimagename+"signed")
+	checkarg(&fileslist, "FILESTOSIGN", "")
+	checkarg(&privkeyfile, "KEYSECRET", "")
+	checkarg(&pubkeyfile, "CERTSECRET", "")
+	checkarg(&pullsecret, "PULLSECRET", "")
+	checkarg(&pushsecret, "PUSHSECRET", pullsecret)
+	// if we've made it this far the arguements are sane
+
+	// get a temp dir to copy kmods into for signing
+	extractiondir, err := os.MkdirTemp("/tmp/", "kmod_signer")
```

Comment:
From the method's doc:
```
It is the caller's responsibility to remove the directory when it is no longer needed.
```

Should we add a `defer` closing function?

Reply:
its running in a containers ephemeral layer so it will get destroyed when that does, but it's always good to tidy up...  I will add it

Reply:
Yes, it might be useful during development where `/tmp` has a slightly longer lifetime.